store/cachemulti/store.go

```diff
@@ -135,9 +140,13 @@ func (cms Store) CacheMultiStore() types.CacheMultiStore {
 
 func (cms Store) getCacheWrap(key types.StoreKey) types.CacheWrap {
 	store, ok := cms.stores[key]
-	if !ok && cms.parentStore != nil {
+	if !ok {
 		// load on demand
-		store = cms.initStore(key, cms.parentStore(key))
+		if cms.branched {
+			store = cms.parentStore(key).(types.BranchStore).Clone().(types.CacheWrap)
```

Comment:
do we check before type cast?

Reply:
no, there's only one cache store implementation, we may cleanup the interfaces in the future though, to better reflect the reality.